.github/workflows/phpunit.yml

```diff
@@ -0,0 +1,283 @@
+          # Ensure MariaDB data directory exists
+          mkdir -p ${mariadb_data_dir}
+          ls -l
+          
+          # Initialize MariaDB if not already done
+          if [ ! -d "${mariadb_data_dir}/mysql" ]; then
+            mysql_install_db --datadir=${mariadb_data_dir}
+            # Start MariaDB temporarily to set up the database
+            # sudo touch ${mariadb_socket}
+            sudo chmod -R 2777 `dirname ${mariadb_socket}`
+            mysqld --datadir=${mariadb_data_dir} --socket=${mariadb_socket} --skip-grant-tables &
+            TEMP_MYSQL_PID=$!
+            sleep 15 # Increased wait time to ensure MariaDB is ready
+            
+            mysqld --verbose --help | grep "socket"
+            ps aux | grep mysqld
+            # Set root password and authentication method
+            echo "Setting root password..."
+            mysql -uroot -S${mariadb_socket} <<EOF
+            FLUSH PRIVILEGES;
+            ALTER USER 'root'@'localhost' IDENTIFIED BY '${root_password}';
+            FLUSH PRIVILEGES;
+          EOF
+          
+            # Verify root password
+            echo "Verifying root password..."
+            mysql -uroot -p${root_password} -S${mariadb_socket} -e "SELECT 1;" || {
+              echo "Error: Root password verification failed."
+              kill $TEMP_MYSQL_PID
+              wait $TEMP_MYSQL_PID
+              exit 1
+            }
+            
+            mysql -uroot -p${root_password} -S${mariadb_socket} -e "CREATE DATABASE IF NOT EXISTS ${moodle_db_name}" || {
+              echo "Error: Failed to create database ${moodle_db_name}."
+              kill $TEMP_MYSQL_PID
+              wait $TEMP_MYSQL_PID
+              exit 1
+            }
+            
+            if [ -f "${moodle_sql_file}" ]; then
+              mysql -uroot -p${root_password} -S${mariadb_socket} ${moodle_db_name} < ${moodle_sql_file} && {
+                echo "SQL file imported successfully."
+              } || {
+                echo "Error: Failed to import SQL file."
+              }
+            else
+              echo "Warning: ${moodle_sql_file} not found. Database created but not populated."
+            fi
+            
+            kill $TEMP_MYSQL_PID
+            wait $TEMP_MYSQL_PID
+          fi
+
+          # Kill existing MariaDB and PHP processes
+          kill_existing "mysqld"
+          kill_existing "php"
+
+          # Start MariaDB
+          start_mariadb() {
+            echo "Starting MariaDB..."
+            mysqld --datadir=${mariadb_data_dir} --socket=${mariadb_socket} &
+            MARIADB_PID=$!
+            sleep 10 # Increased wait time to ensure MariaDB is ready
+
+            # Check if MariaDB socket file is created
+            if [ ! -S ${mariadb_socket} ]; then
+              echo "Error: MariaDB socket file not found at ${mariadb_socket}. Exiting..."
+              exit 1
+            fi
+          }
+
+          # I do not install adminer because I'm lazy
```

Comment:
^^

Reply:
The comment has been fixed to be more explanatory